api/v1/perconaservermysql_types_test.go

```diff
@@ -1,13 +1,61 @@
 package v1
 
 import (
+	"testing"
+
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
 	"github.com/stretchr/testify/assert"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
-	"testing"
 )
 
+func TestCheckNSetDefaults(t *testing.T) {
+	t.Run("empty cr", func(t *testing.T) {
+		cr := new(PerconaServerMySQL)
+		expectedErr := "backup.image can't be empty"
+
+		err := cr.CheckNSetDefaults(t.Context(), nil)
+		if err.Error() != expectedErr {
+			t.Errorf("unexpected err: %s", err.Error())
+		}
+	})
+	t.Run("with backup image", func(t *testing.T) {
+		cr := new(PerconaServerMySQL)
+		cr.Spec.Backup = &BackupSpec{
+			Image: "backup-image",
+		}
+		expectedErr := "reconcile mysql volumeSpec: volumeSpec provided is nil"
+
+		err := cr.CheckNSetDefaults(t.Context(), nil)
+		if err.Error() != expectedErr {
+			t.Errorf("unexpected err: %s", err.Error())
+		}
```

Comment:
Here we can assert the expected error without having to use this error f logic. Similar to the following test case.

Reply:
https://github.com/percona/percona-server-mysql-operator/pull/1072/commits/ddedd145fe5dfbb612bda55ca8373c39f6796f20